src/menu/menu.c

```diff
@@ -1,57 +1,103 @@
 #include <stdio.h>
 #include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
 
-#include <fatfs/ff.h>
 #include <libdragon.h>
 
 #include "flashcart/flashcart.h"
+#include "libs/toml/toml.h"
 
 #include "menu.h"
 
 
-static char *config_path = "n64/config.txt";
-static char game_path[256];
-static char save_path[256];
+#define SC64_CONFIG_FILEPATH "sd://config.sc64.toml.txt"
+static toml_datum_t rom_path;
+static toml_datum_t save_path;
 static flashcart_save_type_t save_type = FLASHCART_SAVE_TYPE_NONE;
 static bool save_writeback = false;
+static bool auto_load_last_rom = false;
 
 
-static void get_string (char *buffer, size_t size, FIL *fil) {
-    UINT br;
-    char c;
+static void load_config (void) {
+    FILE *fp = NULL;
+    char error_buffer[266];
+    
+    printf("Loading config file %s\n", SC64_CONFIG_FILEPATH);
+    wait_ms(1000);
+
+    fp = fopen(SC64_CONFIG_FILEPATH, "r");
+	if (!fp) {
+		printf("Error loading config file %s\n", SC64_CONFIG_FILEPATH);
+        wait_ms(10000);
+        assertf(!fp, "Couldn't open toml config file: %s", SC64_CONFIG_FILEPATH);
```

Comment:
Config loading function definitely shouldn't throw asserts and instead return error code. Menu then could present warning screen for the user if config is broken and default settings were used.